src/jit/hwintrinsicxarch.cpp

```diff
@@ -840,7 +840,7 @@ GenTree* Compiler::impSSE2Intrinsic(NamedIntrinsic        intrinsic,
         {
             assert(sig->numArgs == 1);
             op1      = impSIMDPopStack(TYP_SIMD16);
-            baseType = JITtype2varType(sig->retType);
+            baseType = JITtype2varTypeExact(sig->retType);
```

Comment:
As far as I can tell the intrinsics this code applies to have the following signatures:
```c#
uint ConvertToUInt32(Vector128<uint> value);
ulong ConvertToUInt64(Vector128<ulong> value);
```
So the base type can be obtained from the value parameter like for all other intrinsics. Basically this code seems to be a duplicate of the ConvertToInt32/64 code above.

Reply:
Can we solve this problem by adding `HW_Flag_BaseTypeFromArg`?

Reply:
> Can we solve this problem by adding HW_Flag_BaseTypeFromArg

No it does not work due to logic used in common paths for `HW_Category_SimpleSIMD` or `HW_Category_SIMDScalar`

> the base type can be obtained from the value parameter like for all other intrinsics

Yes, but it is very slow in comparison to `JITtype2varTypeExact`. I do not think it is good practice to use slow code paths in order to avoid very limited change to code base.

Reply:
>  I do not think it is good practice to use slow code paths in order to avoid very limited change to code base.

It's the other way around. It's not good practice to duplicate existing code and add new code as well for something that is not demonstrated to be a performance issue.

These are just 2 intrinsics out of many and they're probably not the most commonly used intrinsics. I'm not sure why they even exist, the signed version would generate the same code.

Reply:
> No it does not work due to logic used in common paths for HW_Category_SimpleSIMD or HW_Category_SIMDScalar

I think all the problem from `ConvertToXXX` can be solved by `HW_Category_SimpleSIMD` + `HW_Flag_BaseTypeFromArg`.